### ISSUE TITLE
tabletop hangs after subscribing to camera topics

### ISSUE THREAD

**Issue by saspelund** [wg-perception/tabletop#30]
I run rosrun `object_recognition_core detection -c  'rospack find object_recognition_tabletop'/conf/detection.object.ros.ork` and it gives me this output before simply stopping

```
/home/<username>/catkin_ws/devel/lib/python2.7/dist-packages/ecto/__init__.py:30: RuntimeWarning: to-Python converter for boost::shared_ptr<ecto::tendril> already registered; second conversion method ignored.
  del os_path
/home/<username>/catkin_ws/devel/lib/python2.7/dist-packages/ecto/__init__.py:30: RuntimeWarning: to-Python converter for boost::shared_ptr<ecto::cell> already registered; second conversion method ignored.
  del os_path
/home/<username>/catkin_ws/devel/lib/python2.7/dist-packages/ecto/__init__.py:30: RuntimeWarning: to-Python converter for boost::shared_ptr<ecto::py::cellwrap> already registered; second conversion method ignored.
  del os_path
[ INFO] [1466695484.352573925]: Initialized ROS. node_name: /object_recognition_server
[ INFO] [1466695484.691118853]: System already initialized. node_name: /object_recognition_server
[ INFO] [1466695485.049567652]: System already initialized. node_name: /object_recognition_server
[ INFO] [1466695485.071410638]: Subscribed to topic:/camera/depth_registered/camera_info [queue_size: 1][tcp_nodelay: 0]
[ INFO] [1466695485.079224758]: Subscribed to topic:/camera/rgb/image_color [queue_size: 1][tcp_nodelay: 0]
[ INFO] [1466695485.092122438]: Subscribed to topic:/camera/depth_registered/image_raw [queue_size: 1][tcp_nodelay: 0]
[ INFO] [1466695485.100964785]: Subscribed to topic:/camera/rgb/camera_info [queue_size: 1][tcp_nodelay: 0]
^C*** SIGINT received, stopping graph execution.
```

The full gdb output including a `where` command can be found [here](https://github.com/saspelund/hello-world/blob/master/tabletopError2016_06_23_11_31_EST)

Any suggestions for fixes?

For what it's worth, I am using the default [conf files](https://github.com/wg-perception/tabletop/tree/master/conf).

Ubuntu 14.04.4 LTS
ROS indigo
opencv 2.4.8 (by using command: `pkg-config --modversion opencv`) but I can't figure out how to upgrade it. I think I got it through ecto-opencv.
microsoft kinect
anything else needed?

**Comment by xMutzelx** [comment]
As promised I will try to answer your problem. 
As it seems tabletop is waiting for data, before it publishes the results. Please tell me different things:

1. How do you start your kinect (roslaunch openni_launch openni.launch)?
2. Do you get a pointcloud in rviz (Is your kinect working)?
3. Please check those this topics:
     - /camera/depth_registered/camera_info
     - /camera/rgb/image_color
     - /camera/depth_registered/image_raw
     - /camera/rgb/camera_info
Needed commands:
`rostopic list` displays all active rostopics
`rostopic hz /topic_name` show the rate the topic is published at
`rostopic echo/topic_name` print the topic incoming data

**Comment by saspelund** [comment]
Thanks for the response!

Here are the commands I run as per the instruction on this [tabletop tutorial](http://wg-perception.github.io/ork_tutorials/tutorial02/tutorial.html):

- `roscore`
- `roslaunch openni_launch openni.launch` - [output](https://github.com/saspelund/hello-world/blob/master/openni_launch_output)
- `rosrun rviz rviz` - [output](https://github.com/saspelund/hello-world/blob/master/rosrun_rviz_output) - [proof of pointcloud working in RViz](http://imgur.com/Cb8mogx)
- `rosrun object_recognition_core detection -c 'rospack find object_recognition_tabletop'/conf/detection.object.ros.ork` - output in previous post

Debugging info:

- `rostopic list` - [output](https://github.com/saspelund/hello-world/blob/master/rostopic_list)
- `rostopic hz /topic` - [output](https://github.com/saspelund/hello-world/blob/master/rostopic_hz_output)
- `rostopic echo -n 1 /topic` - [output](https://github.com/saspelund/hello-world/blob/master/rostopic_echo_output)

I think this may have found the culprit as `rostopic hz /camera/depth_registered/*` and `rostopic echo -n 1 /camera/depth_registered/*`produce no output. I am intrigued as to how RViz point cloud still works..... I am especially intrigued as I was able to create models/meshes of the four objects in the RViz using this kinect. I guess I now have to look into why these two topics produce no output. Thank you for all of your help and I continue to appreciate any additional guidance you may have.

P.S. Sorry if it made it seem like I was singling you out for help on all of these posts. I didn't realize I was asking the same user on all of the posts until I saw my inbox this morning. Apologies.

**Comment by srgarbarino** [comment]
Did you try 

$ rosrun dynamic_reconfigure dynparam set /camera/depth_registration True

You might not have depth_registration set. This can also be accesses through rqt_reconfigure under the camera driver

**Comment by saspelund** [comment]
>.< I thought that was a one-off setting that was a permanent change and so I skipped it after the first time. I feel so stupid some times......

Thank you so very much! I will work with this and see where the next error pops up. I'll close this issue if it all works out fine

**Comment by saspelund** [comment]
So now it runs up to this point:

[ INFO] [1467401594.496830309]: publishing to topic:/recognized_object_array

I can see the marker_array for tabletop/clusters but I can't get the ork_table to show up in RViz; table_array is being published and I can get an output from rostopic echo but it is listed as unvisualizeable in RViz.

**Comment by xMutzelx** [comment]
This output seems to be right. So you can not display the table and the objects are working fine, right?
A stupid question (sorry): Have you added the "OrkTable" in RVIZ?

It should look like this:
**OrkTable**
Topic: /table_array

and for the object detection:
**OrkObject**
Topic: /recognized_object_array

If my answer doesn't fit your question, could you please provide a picture or the exact error message? I am not sure if I understood you right.

**Comment by saspelund** [comment]
I was having the same problem as [this user](https://groups.google.com/forum/#!topic/object-recognition-kitchen/1VEwp7Ww_6A) and [this user](http://answers.ros.org/question/222578/adding-a-new-folder-with-displays-in-rviz/) in that I didn't have the option to add the display type in RViz. I could see the /table_array and /recognized_object_array topics in both Rostopic and under RViz's unvisualizable topics, but I didn't see how to add it.

Over the weekend, I realized that the second link explained that I had to source my setup.bash again which solved the problem. Again, this has to do with me thinking that certain changes carry over from previous sessions/terminals (I didn't know you had to source the setup.bash file for every terminal because doing it once in the parent terminal solved everything before). Now the program is identifying objects like a champ though I need to clear the clutter from my viewspace as the program is working rather hard to deal with all the non-planar objects in the background.

I am new to ROS and this level of computer work in general so this has just been a matter of me downloading packages, waiting for errors, and then searching/asking the internet for help so I greatly appreciate you assistance. Thanks for the response and talk to you soon (undoubtedly)!
